File: ios/HellenicDirectory/Features/Directories/Models/Directory.swift
import Foundation

struct Directory: Decodable, Identifiable {
    let id: String
    let name: String
    let slug: String
    let description: String?
    let isVisible: Bool
    let createdAt: Date
}

struct DirectoryMember: Decodable, Identifiable {
    let id: String
    let directoryId: String
    let userId: String
    let photoUrl: String?
    let city: String?
    let industry: String?
    let employer: String?
    let preferredContact: String?
    let bio: String?
    let joinedAt: Date
    let user: MemberUser
    let organizations: [MemberOrganization]
}

struct MemberUser: Decodable {
    let id: String
    let fullName: String
    let email: String
    let phone: String?
}

struct MemberOrganization: Decodable, Identifiable {
    let id: String
    let organization: Organization
    let verifiedAt: Date?
}

struct Organization: Decodable, Identifiable {
    let id: String
    let name: String
    let type: String?
}

struct PaginatedMembers: Decodable {
    let data: [DirectoryMember]
    let meta: PaginationMeta
}

File: shared/src/types/directory.ts
export type InvitationStatus = "PENDING" | "ACCEPTED" | "REJECTED" | "EXPIRED";

export interface Directory {
  id: string;
  name: string;
  slug: string;
  description?: string;
  isVisible: boolean;
  createdAt: string;
}

export interface Organization {
  id: string;
  name: string;
  type?: string;
  description?: string;
}

export interface MemberOrganization {
  id: string;
  memberId: string;
  organization: Organization;
  verifiedAt?: string;
}

export interface DirectoryMember {
  id: string;
  directoryId: string;
  userId: string;
  photoUrl?: string;
  city?: string;
  industry?: string;
  employer?: string;
  preferredContact?: "email" | "phone" | "in-app";
  bio?: string;
  joinedAt: string;
  user: { id: string; fullName: string; email: string; phone?: string };
  organizations: MemberOrganization[];
}

export interface DirectoryInvitation {
  id: string;
  email: string;
  status: InvitationStatus;
  expiresAt: string;
}
